pylint/lint/pylinter.py

```diff
@@ -938,16 +939,13 @@ def initialize(self):
             if not msg.may_be_emitted():
                 self._msgs_state[msg.msgid] = False
 
-    def check(self, files_or_modules):
+    def check(self, files_or_modules: Sequence[str]) -> None:
         """main checking entry: check a list of files or modules from their name.
 
-        files_or_modules is either a string or list of strings presenting modules to check.
+        files_or_modules is a sequence of strings presenting modules to check.
         """
         self.initialize()
 
-        if not isinstance(files_or_modules, (list, tuple)):
-            files_or_modules = (files_or_modules,)
-
```

Comment:
I like this, clearer input, it will make the code better. But I think this might be a breaking change.

Reply:
Technically all changes of function arguments are breaking ones as well (`check_single_file`, `check`, `_check_files`). It depends if we would consider them part of the public API. Something I'm not sure about.

Reply:
Well I think the check function can be used by addon relying on pylint's checker. I'm for changing the typing only right now and removing the mechanism in 3.0 as a breaking change (we can also add a deprecation warning).